velox/substrait/SubstraitToVeloxExpr.cpp

```diff
@@ -0,0 +1,267 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "SubstraitToVeloxExpr.h"
+#include "TypeUtils.h"
+
+namespace facebook::velox::substrait {
+
+std::shared_ptr<const core::FieldAccessTypedExpr>
+SubstraitVeloxExprConverter::toVeloxExpr(
+    const ::substrait::Expression::FieldReference& sField,
+    const int32_t& inputPlanNodeId) {
+  auto typeCase = sField.reference_type_case();
+  switch (typeCase) {
+    case ::substrait::Expression::FieldReference::ReferenceTypeCase::
+        kDirectReference: {
+      auto dRef = sField.direct_reference();
+      int32_t colIdx = subParser_->parseReferenceSegment(dRef);
+      auto fieldName = subParser_->makeNodeName(inputPlanNodeId, colIdx);
+      // TODO: Get the input type and support different types here.
+      return std::make_shared<const core::FieldAccessTypedExpr>(
+          DOUBLE(), fieldName);
+    }
+    default:
+      VELOX_NYI(
+          "Substrait conversion not supported for Reference '{}'", typeCase);
+  }
+}
+
+std::shared_ptr<const core::ITypedExpr>
+SubstraitVeloxExprConverter::toVeloxExpr(
+    const ::substrait::Expression::ScalarFunction& sFunc,
+    const int32_t& inputPlanNodeId) {
+  std::vector<std::shared_ptr<const core::ITypedExpr>> params;
+  for (const auto& sArg : sFunc.args()) {
+    auto expr = toVeloxExpr(sArg, inputPlanNodeId);
+    params.push_back(expr);
+  }
+  auto functionId = sFunc.function_reference();
+  auto veloxFunction = subParser_->findVeloxFunction(functionMap_, functionId);
+  auto subType = subParser_->parseType(sFunc.output_type());
+  auto veloxType = toVeloxType(subType->type);
+  return std::make_shared<const core::CallTypedExpr>(
+      veloxType, std::move(params), veloxFunction);
+}
+
+std::shared_ptr<const core::ConstantTypedExpr>
+SubstraitVeloxExprConverter::toVeloxExpr(
+    const ::substrait::Expression::Literal& sLit) {
+  auto typeCase = sLit.literal_type_case();
+  switch (typeCase) {
+    case ::substrait::Expression_Literal::LiteralTypeCase::kFp64: {
```

Comment:
nit: don't need the code block here '{}'

Reply:
Fixed the cases like this.